server/test/repository/VersionRepositoryTest.java

```diff
@@ -37,38 +42,84 @@ public void testPublish() {
     resourceCreator.insertActiveProgram("foo");
     resourceCreator.insertActiveProgram("bar");
     resourceCreator.insertDraftProgram("bar");
-    assertThat(this.versionRepository.getActiveVersion().getPrograms()).hasSize(2);
-    assertThat(this.versionRepository.getDraftVersion().getPrograms()).hasSize(1);
-    Version oldDraft = this.versionRepository.getDraftVersion();
-    this.versionRepository.publishNewSynchronizedVersion();
-    assertThat(this.versionRepository.getActiveVersion().getPrograms()).hasSize(2);
-    assertThat(this.versionRepository.getDraftVersion().getPrograms()).hasSize(0);
+    assertThat(versionRepository.getActiveVersion().getPrograms()).hasSize(2);
+    assertThat(versionRepository.getDraftVersion().getPrograms()).hasSize(1);
+    Version oldDraft = versionRepository.getDraftVersion();
+    versionRepository.publishNewSynchronizedVersion();
+    assertThat(versionRepository.getActiveVersion().getPrograms()).hasSize(2);
+    assertThat(versionRepository.getDraftVersion().getPrograms()).hasSize(0);
     oldDraft.refresh();
     assertThat(oldDraft.getLifecycleStage()).isEqualTo(LifecycleStage.ACTIVE);
   }
 
+  @Test
+  public void testPublishDoesNotUpdateProgramTimestamps() throws InterruptedException {
+    ImmutableList<Program> programs =
+        ImmutableList.of(
+            resourceCreator.insertActiveProgram("active"),
+            resourceCreator.insertActiveProgram("other_active"),
+            resourceCreator.insertActiveProgram("active_with_draft"),
```

Comment:
Done, with a slight reservation. The new order reflects a state we never intend the system to be in: There's a program that's in "Draft" with an earlier created timestamp than an "Active" program.

That said, it's probably not actually a big deal since this test purely asserts that the updated timestamp didn't change.

Reply:
I'm not sure I follow.

There's 4 programs here 1 of which has 2 revisions.  the ordering of the 4 can be any order.

The one with 2 revisions seems to be in correct order, the Active first, the Draft second.